int64[] shovel_angles